lib/benchmarks/src/Cardano/Wallet/Benchmarks/History.hs

```diff
@@ -0,0 +1,187 @@
+{-# LANGUAGE DerivingStrategies #-}
+module Cardano.Wallet.Benchmarks.History
+    ( HarmonizedHistory (..)
+    , History
+    , IndexedSemantic (..)
+    , HarmonizedRow (..)
+    , harmonizeHistory
+    , pastDays
+    , renderHarmonizedHistoryCsv
+    )
+where
+
+import Prelude
+
+import Cardano.Wallet.Benchmarks.Collect
+    ( Result (Result, resultUnits)
+    , Semantic
+    , Units (..)
+    , convertUnits
+    )
+import Data.Bifunctor
+    ( second
+    )
+import Data.Csv
+    ( Header
+    , ToNamedRecord (..)
+    , namedRecord
+    , (.=)
+    )
+import Data.Foldable
+    ( Foldable (..)
+    )
+import Data.Functor
+    ( (<&>)
+    )
+import Data.Map
+    ( Map
+    )
+import Data.Map.Monoidal.Strict
+    ( MonoidalMap (..)
+    , assocs
+    )
+import Data.Semigroup
+    ( First (..)
+    )
+import Data.Set
+    ( Set
+    )
+import Data.Time
+    ( Day
+    , UTCTime (..)
+    , addDays
+    , getCurrentTime
+    )
+
+import qualified Data.ByteString.Char8 as B8
+import qualified Data.Map as Map
+import qualified Data.Map.Monoidal.Strict as MMap
+import qualified Data.Text as T
+import qualified Data.Vector as V
+
+data IndexedSemantic = IndexedSemantic
+    { semantic :: Semantic
+    , index :: Int
+    }
+    deriving stock (Eq, Ord, Show)
+
+type History = MonoidalMap IndexedSemantic (MonoidalMap Day (First Result))
+
+data HarmonizedRow
```

Comment:
Could you add a comment to the data type describing what it represents?

E.g. "A row of measured values, where all values have been harmonized to the same unit."